qiskit/algorithms/optimizers/utils/learning_rate.py

```diff
@@ -0,0 +1,83 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A class to represent the Learning Rate."""
+
+from typing import Union, Callable, Optional, List, Iterator, Generator
+from itertools import tee
+import numpy as np
+
+
+class LearningRate(Generator):
+    """Represents a Learning Rate.
+    Will be an attribute of :class:`~.GradientDescentState`. Note that :class:`~.GradientDescent` also
+    has a learning rate. That learning rate can be a float, a list, an array, a function returning
+    a generator and will be used to create a generator to be used during the
+    optimization process.
+    This class wraps ``Generator`` so that we can also access the last yielded value.
+    """
+
+    def __init__(
+        self, learning_rate: Union[float, List[float], np.ndarray, Callable[[], Iterator]]
+    ):
+        """
+        Args:
+            learing_rate: Used to create a generator to iterate on.
```

Comment:
Typo in the arg here
```suggestion
            learning_rate: Used to create a generator to iterate on.
```